src/frontend/src/flows/manage.ts

```diff
@@ -17,86 +17,138 @@ import { WebAuthnIdentity } from "@dfinity/identity";
 import { setupRecovery } from "./recovery/setupRecovery";
 import { hasOwnProperty } from "../utils/utils";
 
-const pageContent = (userNumber: bigint, devices: DeviceData[]) => html`<style>
-    .nagBox {
-      display: flex;
-      align-items: center;
-      gap: 1rem;
-      padding: 1rem;
-      margin-bottom: 2rem;
-      box-sizing: border-box;
-      border-style: double;
-      border-width: 2px;
-      border-radius: 4px;
-      border-image-slice: 1;
-      outline: none;
-      border-image-source: linear-gradient(
-        270.05deg,
-        #29abe2 10.78%,
-        #522785 22.2%,
-        #ed1e79 42.46%,
-        #f15a24 59.41%,
-        #fbb03b 77.09%
-      );
-    }
-    .nagIcon {
-      align-self: flex-start;
-    }
-    .recoveryNag {
-      display: flex;
-      flex-direction: column;
-    }
-    .recoveryNagTitle {
-      font-weight: 600;
-      font-size: 1.1rem;
-    }
-    .recoveryNagMessage {
-      margin-top: 0.5rem;
-      margin-bottom: 1rem;
-      font-size: 1rem;
-    }
-    .recoveryNagButton {
-      padding: 0.2rem 0.4rem;
-      border-radius: 2px;
-      width: fit-content;
-      align-self: flex-end;
-      margin: 0;
-    }
-    .labelWithAction {
-      margin-top: 1rem;
-      display: flex;
-      justify-content: space-between;
-    }
+// The various error messages we may display
 
-    .labelWithAction label {
-      margin: 0;
-    }
+const displayFailedToAddNewDevice = (error: Error) =>
+  displayError({
+    title: "Failed to add new device",
+    message: html`
+      We failed to add your new device.<br />
+      If you're trying to add a device that is not attached to this machine try
+      following the instructions at<br />
+      <a
+        target="_blank"
+        href="https://sdk.dfinity.org/docs/ic-identity-guide/auth-how-to.html#_add_a_device"
+        >https://sdk.dfinity.org/docs/ic-identity-guide/auth-how-to.html#_add_a_device</a
+      >
+    `,
+    detail: error.message,
+    primaryButton: "Back to manage",
+  });
 
-    .labelAction {
-      padding: 0;
-      border: none;
-      display: inline;
-      width: auto;
-      margin: 0;
-      cursor: pointer;
-      color: #387ff7;
-      font-size: 12px;
-      font-family: "Montserrat", sans-serif;
-      text-align: right;
-      font-weight: 600;
-    }
-    .labelAction::before {
-      content: "+";
-      margin-right: 3px;
-      color: #387ff7;
-    }
-  </style>
+const displayFailedToAddTheDevice = (error: Error) => {
+  displayError({
```

Comment:
You need to either `return` the Promise or remove the surrounding block here, or you can't `await` the user clicking "Back to manage"